sw/host/opentitanlib/src/bootstrap/legacy.rs

```diff
@@ -0,0 +1,287 @@
+                        // Note: boot_rom only likes 0 and 0x40000 as start addresss, but
+                        // we do not enforce that here.
+                        addr = skip_addr;
+                    } else if addr & 31 != 0 {
+                        // Only skip if not left at exact 128 multiple, per boot_rom.
+                        addr = skip_addr;
+                    }
+                }
+            }
+            
+            if addr + Self::DATA_LEN >= max_addr {
+                frameno |= Self::EOF;
+            }
+
+            let mut frame = Frame {
+                header: FrameHeader {
+                    frame_num: frameno,
+                    flash_offset: (addr + Self::FLASH_BASE_ADDRESS) as u32,
+                    ..Default::default()
+                },
+                ..Default::default()
+            };
+            let slice_size = std::cmp::min(Self::DATA_LEN, payload.len() - addr);
+            frame.data[..slice_size].copy_from_slice(&payload[addr..addr + slice_size]);
+            frame.header.hash = frame.header_hash();
+            frames.push(frame);
+
+            
+            addr += Self::DATA_LEN;
+            frameno += 1;
+        }
+        frames
+    }
+}
+
+/// Implements the bootstrap protocol of previous Google Titan family chips.
+pub struct Legacy {
+    pub inter_frame_delay: Duration,
+    pub flash_erase_delay: Duration,
+}
+
+impl Legacy {
+    const INTER_FRAME_DELAY: Duration = Duration::from_millis(20);
+    const FLASH_ERASE_DELAY: Duration = Duration::from_millis(200);
+
+    /// Creates a new `Primitive` protocol updater from `options`.
+    pub fn new(options: &BootstrapOptions) -> Self {
+        Self {
+            inter_frame_delay: options.inter_frame_delay.unwrap_or(Self::INTER_FRAME_DELAY),
+            flash_erase_delay: options.flash_erase_delay.unwrap_or(Self::FLASH_ERASE_DELAY),
+        }
+    }
+}
+
+#[derive(Debug, Error)]
+pub enum LegacyBootstrapError {
+    #[error("Boot rom not ready")]
+    NotReady(),
+    #[error("Unknown boot rom error: {0}")]
+    Unknown(u8),
+    #[error("Boot rom error: NOREQUEST")]
+    NOREQUEST(),
+    #[error("Boot rom error: NOMAGIC")]
+    NOMAGIC(),
+    #[error("Boot rom error: TOOBIG")]
+    TOOBIG(),
+    #[error("Boot rom error: TOOHIGH")]
+    TOOHIGH(),
+    #[error("Boot rom error: NOALIGN")]
+    NOALIGN(),
+    #[error("Boot rom error: NOROUND")]
+    NOROUND(),
+    #[error("Boot rom error: BADKEY")]
+    BADKEY(),
+    #[error("Boot rom error: BADSTART")]
+    BADSTART(),
+    #[error("Boot rom error: NOWIPE")]
+    NOWIPE(),
+    #[error("Boot rom error: NOWIPE0")]
+    NOWIPE0(),
+    #[error("Boot rom error: NOWIPE1")]
+    NOWIPE1(),
+    #[error("Boot rom error: NOTEMPTY")]
+    NOTEMPTY(),
+    #[error("Boot rom error: NOWRITE")]
+    NOWRITE(),
+    #[error("Boot rom error: BADADR")]
+    BADADR(),
+    #[error("Boot rom error: OVERFLOW")]
+    OVERFLOW(),
```

Comment:
I think the rust naming convention for these is UpperCamelCase.  I'd expect the compiler or clippy to complain about these names.

Reply:
Done.